core/DataAccess/LogAggregator.php

```diff
@@ -219,7 +225,7 @@ public function getSegmentTmpTableName()
 
     public function cleanup()
     {
-        if (!$this->segment->isEmpty() && $this->isSegmentCacheEnabled()) {
+        if (!$this->segment->isEmpty() && ($this->isSegmentCacheEnabled() || $this->forceCheckForSegmentTmpOnCleanup)) {
```

Comment:
is this property `forceCheckForSegmentTmpOnCleanup` still needed?

Reply:
No, I should have removed this with the previous rework - fixed.